turbopack/crates/turbopack-ecmascript/src/references/esm/binding.rs

```diff
@@ -149,46 +149,37 @@ impl EsmBinding {
                     ));
                     break;
                 }
-                Some(swc_core::ecma::visit::AstParentKind::BindingIdent(
-                    swc_core::ecma::visit::fields::BindingIdentField::Id,
-                )) => {
+                // We need to handle LHS because of code like
+                // (function (RouteKind1){})(RouteKind || RouteKind = {})
+                Some(swc_core::ecma::visit::AstParentKind::SimpleAssignTarget(_)) => {
                     ast_path.pop();
 
```

Comment:
We fall back to `ast_path.pop()` if we encounter an unknown node, so no need to do it here.